### DISCUSSION TITLE
homematic.keypress für HmIP-FCI1

### DISCUSSION THREAD

**Discussion by emufan** [SukramJ/aiohomematic#811] [Q&A] [answered]
Irgendwie mache ich gerade alles falsch oder anders als sonst, aber ich bekomme einfach 0,0 Events von einem HmIP-FCI1. 

Diesen habe ich dafür noch nicht benutzt, aber ich habe es versucht, wie bei allen anderen Solo-Tastern oder Tastern auf Aktoren auch. Dummyprogramm auf CCU. Und spätestens danach (kA ob das immer noch notwendig ist, mache es aus Gewohnheit immer (noch)) kamen die Events ins HA.

Aber vom HmIP-FCI1 leider gar keine. Nicht im Listener und logischer Weise schon gar nicht als Trigger in Automatisierungen. 

Das Gerät zeigt mir auch die Optionen an
![image](https://user-images.githubusercontent.com/22775515/210530686-0f0d00c3-dac6-4ffd-b6e3-b82e1ff0a328.png)
wobei ich hier nicht weiß, ob die erst erscheinen, wenn aktiviert (oben) oder je nach Gerättyp immer, selbst wenn ohne Funktion.

Derzeit hat derHmIP-FCI1 eine Direktverknüpfung (kur) auf einem anderen Aktor. Aber ich würde hoffen, dass dies nicht das Problem ist.

Hat jemand eine Idee, was hier falsch läuft?

**Comment by SukramJ** [comment]
> wobei ich hier nicht weiß, ob die erst erscheinen, wenn aktiviert (oben) oder je nach Gerättyp immer, selbst wenn ohne Funktion.

Die Optionen erscheinen immer.

**Reply by emufan** [reply]
Danke für die Info. Werde ich mir merken. Hilft also schon einmal nicht für meine Problemfindung.

**Comment by emufan** [comment]
OT für diese Frage. Warum ist die Event-Abfolge bei langem Festhalten eigentlich

press_long
press_long_start
press_long
press_long
press_long
press_long
press_long_release

d.h. warum kommt (bei meinen Tests immer) ein zusätzliches ein long vor dem long_start von der CCU. Für meine Fälle bisher kein Problem, aber man will ja nicht dumm bleiben und es verstehen. Geht ja nur um ms, aber eigentlich wäre dann ja ein Reagieren auf press_long schneller als auf press_long_start

**Reply by danielperna84** [reply]
Ich vermute, dass das die neue Form der Implementierung von [PRESS_CONT](https://github.com/danielperna84/pydevccu/search?p=1&q=PRESS_CONT), welche es bei den alten HM-Geräten gab. Da war's (glaube ich) so:
- PRESS_LONG
- PRESS_CONT
- ...
- PRESS_CONT
- PRESS_LONG_RELEASE

Dieser Ansatz hatte den Nachteil, dass man z.B. nicht mit `PRESS_LONG` mit jedem Event etwas ausführen konnte (weil's _long_ ja nur ein mal gibt). Und wenn man nur auf `PRESS_CONT` hört, dann ist's verzögert weil das erste `PRESS_LONG` ja nicht mit gezählt wird.

Beim neuen Ansatz nimmt man einfach immer `PRESS_LONG` und es funktioniert wie man es erwartet.

Ist aber nur eine Vermutung.

**Reply by emufan** [reply]
Danke für die Info. Dann bin ich wieder etwas schlauer. Was bedeutet das 

> weil das erste PRESS_LONG ja nicht mit gezählt wird

press_long ist halt meist unschön für Trigger in HA, da dann diese unschönen Meldungen "Automatisierung läuft schon" erscheinen, wenn ich wie meist single und ein delay am Ende in der Automatisierung hatte. Daher gefällt mir _start besser, dann braucht es kein single und kein Delay. Und daber ist mir dann das _long vorher aufgefallen, als ich das mal irgendwann umgestellt hatte.

**Reply by danielperna84** [reply]
Damit meine ich, dass man bei den alten Geräten nicht nur `PRESS_LONG` als Trigger nehmen kann, wenn man damit (auch) mit jedem weiteren `PRESS_CONT` dieselbe Aktion ausführen will. Hier geht's ja z.B. um sowas wie Lichter dimmen. Da will man mit jedem Event die Helligkeit erhöhen oder verringern. Wenn man also nur `PRESS_LONG` nimmt, dann wird die Helligkeit nur ein Mal erhöht, denn das Event gibt's nur ein Mal. Um sie kontinuierlich zu erhöhen muss man also zusätzlich auch noch auf `PRESS_CONT` reagieren.

Bei den neuen Geräten ist es einfach nur noch `PRESS_LONG` als Trigger, und falls für den User relevant, kann er mit `PRESS_LONG_START` erkennen, dass es >1 `PRESS_LONG` Events sind.

Nachtrag (um auf die eigentliche Frage einzugehen): damit war eine Automation gemeint, bei der man nur auf `PRESS_CONT` reagiert. Weil das erste Event `PRESS_LONG` ist, würde im Dimmer-Beispiel die Helligkeit erst verzögert geändert werden, denn das erste Event ist ja kein `PRESS_CONT` sondern `PRESS_LONG`.

**Reply by emufan** [reply]
Ao. Ok. Danke.

Wollte nur meinen Use-Case (2te-separate Einzel-Funktion per HA mit langem Tastendruck getriggert) genannt haben. Dafür ist `PRESS_LONG_START` besser.

**Comment by SukramJ** [comment]
Also wenn Du die relevanten Events in einem Dummy Programm verwendest, trotzdem keine Events in HA ankommen, und diese auch nicht in der CCU/RM protokolliert werden, dann denke ich das diese Frage besser bei [RaspberryMatic](https://github.com/jens-maus/RaspberryMatic/discussions) gestellt wird.
Häufig kommen bei nicht erwartetem Geräteverhalten Antworten wie:
- Gerät auf Werkszustand zurücksetzen
- Gerät ab- und wieder anmelden

**Comment by emufan** [comment, marked as answer]
Hab' zwar nicht auf Werkzustand zurückgesetzt, aber nun mal alles nochmal durchgestartet. Warum auch immer (habe nichts geändert) stand nun eine Aktualisierung in der CCU zu dem Gerät aus. Nachdem ich das erledigt hatte, geht es nun auch wie gewünscht. Woran es nun lag ... keine Ahnung.
